### PR TITLE
Sync SHAKE256 default (disabled) with parent default edDSA448 (disabled) and remove WOLFSSL_NO_SHAKE256 flag

### PR DESCRIPTION
**Pull request by kaleb-himes**
The goal of this PR is to establish a fixed default setting for SHAKE256 that is consistent with the feature that depends on it. ed448 (specifically edDSA448) requires shake256 but ed448 is off by default so it did not make sense to have shake256 on by default.

It is also intended to remove checks for both enabled and disabled such as `#if defined(WOLFSSL_SHAKE256) && !defined(WOLFSSL_NO_SHAKE256)`, we have no other features like this. We should only have a default setting and a switch to toggle the non-default state.

Based on ed448 being disabled by default the fix here is to make shake256 also disabled by default and to make it consistent with other hash algorithms that are disabled by default such as sha512 which requires a user to define WOLFSSL_SHA512 to enable sha512, similarly users should only need to toggle one setting for SHAKE256. If user has not defined WOLFSSL_SHAKE256 then the feature should be off.

### CODE REVIEW

**Review by embhorn — APPROVED** [review]
Excellent! Thanks, Kaleb.

File: configure.ac [hunk 1]
```diff
@@ -1776,9 +1776,8 @@ fi
 
 # SHAKE256
 AC_ARG_ENABLE([shake256],
-    [AS_HELP_STRING([--enable-shake256],[Enable wolfSSL SHAKE256 support (default: enabled on x86_64/aarch64)])],
-    [ ENABLED_SHAKE256=$enableval ],
-    [ ENABLED_SHAKE256=$ENABLED_SHA3 ]
```
Comment (SparkiDev) [inline configure.ac:1781]: SHAE256 is inly available if SHA-3 is compiled in.
Make sure SHA-3 is compiled in when SHAKE256 is required.

**Review by SparkiDev — CHANGES_REQUESTED** [review]

File: configure.ac (hunk 1 above)
Comment (SparkiDev) [reply configure.ac:1781]: Maybe put a warning in somewhere if SHAKE256 is enabled by SHA-3 not.

Comment (kaleb-himes) [reply configure.ac:1781]: Is the existing check somehow insufficient?

https://github.com/wolfSSL/wolfssl/pull/4080/files#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R3211-R3214

Comment (SparkiDev) [reply configure.ac:1781]: Check in there on line 3211 of configure.ac.

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**